src/Data/OpenApi/Internal.hs

```diff
@@ -1425,6 +1445,23 @@ instance ToJSON Callback where
 -- Manual FromJSON instances
 -- =======================================================================
 
+instance FromJSON OpenApiSpecVersion where
+  parseJSON = withText "OpenApiSpecVersion" $ \str ->
+            let validatedVersion :: Either String Version
+                validatedVersion = do
+                  parsedVersion <- readVersion str 
+                  upperBound <- readVersion "3.0.3" -- Latest known version that works with the spec
+                  lowerBound <- readVersion "3.0.0"
```

Comment:
Let's use [`makeVersion`](https://hackage.haskell.org/package/base-4.17.0.0/docs/Data-Version.html#v:makeVersion) for these static versions?

Reply:
Addressed in my last commit